src/Acuminator/Acuminator.Analyzers/FixProviders/DAC/ForbiddenFieldsInDacFix.cs

```diff
@@ -0,0 +1,148 @@
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.CodeAnalysis.Editing;
+using Acuminator.Utilities;
+using Microsoft.CodeAnalysis.CSharp.Formatting;
+
+namespace Acuminator.Analyzers.FixProviders
+{
+    [Shared]
+    [ExportCodeFixProvider(LanguageNames.CSharp)]
+    public class ForbiddenFieldsInDacFix : CodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+            ImmutableArray.Create(Descriptors.PX1027_ForbiddenFieldsInDacDeclaration.Id);
+
+        public override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            return Task.Run(() =>
+            {
+                var diagnostic = context.Diagnostics.FirstOrDefault(d => d.Id == Descriptors.PX1027_ForbiddenFieldsInDacDeclaration.Id);
+
+                if (diagnostic == null || context.CancellationToken.IsCancellationRequested)
+                    return;
+
+                    
+                string codeActionName = nameof(Resources.PX1027Fix).GetLocalized().ToString();
+                CodeAction codeAction = CodeAction.Create(codeActionName,
+                                                          cToken => DeleteForbiddenFieldsAsync(context.Document, context.Span, cToken),
+                                                          equivalenceKey: codeActionName);
+
+                context.RegisterCodeFix(codeAction, context.Diagnostics);
+            }, context.CancellationToken);
+        }
+
+        private async Task<Document> DeleteForbiddenFieldsAsync(Document document, TextSpan span, CancellationToken cancellationToken)
+        {
+            SyntaxNode root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            SyntaxNode diagnosticNode = root?.FindNode(span);
+           
+
+            if (diagnosticNode == null || cancellationToken.IsCancellationRequested)
+                return document;
+            ClassDeclarationSyntax classDeclaration = diagnosticNode.Parent<ClassDeclarationSyntax>();
+
+            var rewriterWalker = new RegionClassRewriter((diagnosticNode is ClassDeclarationSyntax) ?
+                                                                (diagnosticNode as ClassDeclarationSyntax)?.Identifier.Text :
+                                                                (diagnosticNode as PropertyDeclarationSyntax)?.Identifier.Text,
+                                                         cancellationToken);
+            var classModified = rewriterWalker.Visit(classDeclaration);
+            
+            if (cancellationToken.IsCancellationRequested)
+                return document;
+            //Format tabulations
+            AdhocWorkspace workspace = new AdhocWorkspace();
+            OptionSet options = workspace.Options; 
+            options = options.WithChangedOption(FormattingOptions.UseTabs,LanguageNames.CSharp,false)
```

Comment:
@saloponov, this is certainly not good idea to use your own tab and indentation settings. What if the user has some other values? Especially for use tabs option - we actually have true in Acumatica for this

Reply:
@saloponov, at least you need to obtain formatting options from the current workspace:
`document.Project.Solution.Workspace. `

Actually you can pass the reference to the WorkSpace to Formatter:
`var formattedResult = Formatter.Format(syntaxNode, workspace);`

Reply:
In my VS default settings set tabulation as "\t" symbol and tests framework can't compare two strings correctly. I will try to use document.Project.Solution.Workspace. Thanks!